### PR TITLE
README: clarify `extraEmacsPackages` usage

### PR DESCRIPTION
**Pull request by acid-bong**
Inspired by [`mu4e`'s recipe](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/emacs/elisp-packages/manual-packages/mu4e/package.nix#L13)

Now here's a lil issue i'd love to resolve:
- if you `nix build|run` such a closure, `extraPackages` won't be available
- only when this closure is installed (with `nix profile install` or adding it to `environment.systemPackages`/`home.packages`), `extraPackages` can be propagated.\*

Is this behaviour fine as is or should I make `extraPackages` always available?

\*Just tried with `nixos-rebuild test`, somehow this doesn't work :thinking:

### CODE REVIEW

**Comment by talyz** [comment]
Another issue is that this only works when `defaultInitFile` is used, which it isn't by default.

**Review by jian-lin — CHANGES_REQUESTED** [review]
I am strongly against using `propagatedUserEnvPkgs` **here**.  The reasons are as follows:
- `propagatedUserEnvPkgs` has issues. See links in https://github.com/NixOS/nixpkgs/issues/347093.
- **Here**, we use `emacsWithPackages` which will [add] `bin` dirs of packages in `extraEmacsPackages` to `exec-path`.  To find executables, Emacs lisp packages should respect `exec-path` by using functions like `executable-find`.  If one Emacs lisp package do not respect `exec-path`, that package itself should be fixed.
- I use `propagatedUserEnvPkgs` in the `mu4e` package only to support **unrecommended** use cases where `emacsWithPackages` is not used.  For example, users install Emacs lisp packages by adding them to `environment.systemPackages`.

[adds]: https://github.com/NixOS/nixpkgs/blob/0ac0220f7d2186b1ab81da97e195a45f05ab7f7f/pkgs/applications/editors/emacs/build-support/wrapper.nix#L171

**Comment by acid-bong** [comment]
Ight, I'm limiting this edit only to README, where i clarify that one can just use `extraEmacsPackages` or `.withPackages`

File: README.org [hunk 1]
```diff
@@ -122,13 +122,16 @@ required in a user's config via =use-package= or =leaf=.
         alwaysTangle = true;
 
         # Optionally provide extra packages not in the configuration file.
-        extraEmacsPackages = epkgs: [
+        # This can also include extra executables to be run by Emacs (linters,
+        # language servers, formatters, etc)
+        extraPackages = epkgs: [
           epkgs.cask
+          pkgs.shellcheck
         ];
```
Comment (jian-lin) [inline README.org:130]: Note that renaming `extraEmacsPackages` to `extraPackages` is a **breaking change**.

Adding comment alone is good and `extraEmacsPackages` may not be a good name with the added comment.  However, I do not think the breaking change is justified.  

WDYT?

File: README.org [hunk 2]
```diff
@@ -122,13 +122,16 @@ required in a user's config via =use-package= or =leaf=.
...
 
         # Optionally override derivations.
         override = final: prev: {
-          weechat = prev.melpaPackages.weechat.overrideAttrs(old: {
+          weechat = prev.melpaPackages.weechat.overrideAttrs (old: {
```
Comment (jian-lin) [inline README.org:134]: I would not add unrelated change.

**Review by jian-lin — CHANGES_REQUESTED** [review]

File: README.org (hunk 1 above)
Comment (acid-bong) [reply README.org:130]: Ok, I'll leave the README edit, and the option rename is up to the maintainers

**Review by jian-lin — APPROVED** [review]
LGTM, thanks.
